src/Features/Core/Portable/ExtractMethod/MethodExtractor.CodeGenerator.cs

```diff
@@ -85,13 +87,20 @@ public async Task<GeneratedCode> GenerateAsync(CancellationToken cancellationTok
                 var codeGenerationService = SemanticDocument.Document.GetLanguageService<ICodeGenerationService>();
                 var result = GenerateMethodDefinition(LocalFunction, cancellationToken);
 
+                var (expressionBodiedMethod, expressionBodiedLocalFunction, staticLocalFunction) = await StaticLocalFunctionAndExpressionBodyPreferenceAsync(SemanticDocument, cancellationToken).ConfigureAwait(false);
+
                 SyntaxNode destination, newContainer;
                 if (LocalFunction)
                 {
                     destination = InsertionPoint.With(callSiteDocument).GetContext();
                     var localMethod = codeGenerationService.CreateMethodDeclaration(
                         method: result.Data,
-                        options: new CodeGenerationOptions(generateDefaultAccessibility: false, generateMethodBodies: true, parseOptions: destination?.SyntaxTree.Options));
+                        options: new CodeGenerationOptions(
+                            generateDefaultAccessibility: false,
+                            generateMethodBodies: true,
+                            preferExpressionBodiedLocalFunction: expressionBodiedLocalFunction,
+                            preferStaticLocalFunction: staticLocalFunction,
+                            parseOptions: destination?.SyntaxTree.Options));
```

Comment:
preferences should likely be allowed by just having CodeGenerationOptions take in an OptionSet.  Then the final code can figure this out based on user settings.